Lib9c/Action/DailyReward.cs

```diff
@@ -29,23 +29,38 @@ public class DailyReward : GameAction, IDailyRewardV1
         public override IAccountStateDelta Execute(IActionContext context)
         {
             var states = context.PreviousStates;
-            var inventoryAddress = avatarAddress.Derive(LegacyInventoryKey);
-            var worldInformationAddress = avatarAddress.Derive(LegacyWorldInformationKey);
-            var questListAddress = avatarAddress.Derive(LegacyQuestListKey);
             if (context.Rehearsal)
             {
                 return states
                     .SetState(avatarAddress, MarkChanged)
-                    .SetState(inventoryAddress, MarkChanged)
-                    .SetState(worldInformationAddress, MarkChanged)
-                    .SetState(questListAddress, MarkChanged)
                     .MarkBalanceChanged(GoldCurrencyMock, avatarAddress);
             }
 
             var addressesHex = GetSignerAndOtherAddressesHex(context, avatarAddress);
             var started = DateTimeOffset.UtcNow;
             Log.Debug("{AddressesHex}DailyReward exec started", addressesHex);
-            if (!states.TryGetAgentAvatarStatesV2(context.Signer, avatarAddress, out _, out AvatarState avatarState, out _))
+            if (!states.TryGetState(avatarAddress, out Dictionary serializedAvatar))
+            {
+                throw new FailedLoadStateException(
+                    $"{addressesHex}Aborted as the avatar state of the signer was failed to load.");
+            }
+
+            Address? agentAddress = null;
+            bool leagcy = false;
```

Comment:
it seems that this `legacy` doesn't mean the same meaning as `LegacyAgentAddressKey`. if so, how about renaming more informational things like... `legacyApKey`?

Reply:
i'll rename `useLegacyKey`.